fe/fe-core/src/main/java/com/starrocks/analysis/LargeStringLiteral.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.analysis;
+
+import com.google.common.base.Preconditions;
+import com.starrocks.sql.parser.NodePosition;
+
+/**
+ * For explain or profile, we only need to display a portion of the string content.
+ * When converting to SQL text, the complete content is required.
+ */
+public class LargeStringLiteral extends StringLiteral {
+
+    public static final int LEN_LIMIT = 50;
+
+    private String shortSqlStr;
+
+    public LargeStringLiteral(String value, NodePosition pos) {
+        super(value, pos);
+    }
+
+    public String toFullSqlImpl() {
+        return super.toSqlImpl();
+    }
+
+    @Override
+    public String toSqlImpl() {
+        if (shortSqlStr == null) {
+            String fullSql = toFullSqlImpl();
+            Preconditions.checkState(value.length() > LEN_LIMIT);
+            fullSql = fullSql.substring(0, LEN_LIMIT);
```

Comment:
I don't think the checkState is necessary and it will cause unknown error.
You can get a length = min(LEN_LIMIT, value.length())

Reply:
My concern is that view implementation depends on this function, and you change the result of toSql, I'm worried this change will make some views can't work well after this change.

Reply:
We had only two functions to build sql which is `AstToSQLBuilder.visitLiteral()` and `AstToStringBuilder.visitLiteral()`. I had used `toFullSqlImpl()` to ensure `LargeStringLiteral` can return right sql text.